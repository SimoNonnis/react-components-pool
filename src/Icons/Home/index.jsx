import React from 'react';
import PropTypes from 'prop-types';

/**
 * Home Icon.
 * @param {number} width
 * @param {number} height
 * @param {string} fill
 * @param {string} className
 * @param {string} classNamePath
 * @returns {React.Component}
 */
function Home({ width, height, fill, className, classNamePath }) {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 100 100"
      width={width}
      height={height}
      className={className}
    >
      <title>Home Icon</title>

      <path
        className={classNamePath}
        fill={fill}
        d="M98.26,42.6a2.87,2.87,0,0,0-1.15-1.89L51.71,7.46a2.88,2.88,0,0,0-3.41,0L2.89,40.71A2.89,2.89,0,1,0,6.3,45.37l43.7-32,43.7,32a2.87,2.87,0,0,0,1.7.56,2.9,2.9,0,0,0,2.33-1.18A2.87,2.87,0,0,0,98.26,42.6Z"
      />
      <path
        className={classNamePath}
        fill={fill}
        d="M84.84,45.21A2.89,2.89,0,0,0,82,48.1V87.32H62.84V66.09a12.85,12.85,0,0,0-25.69,0V87.32H18V48.1a2.89,2.89,0,0,0-5.78,0V90.21a2.89,2.89,0,0,0,2.89,2.89H40a2.9,2.9,0,0,0,2.88-2.65,2.26,2.26,0,0,0,0-.24V66.09a7.07,7.07,0,1,1,14.14,0V90.21a2.1,2.1,0,0,0,0,.22A2.9,2.9,0,0,0,60,93.1H84.84a2.89,2.89,0,0,0,2.89-2.89V48.1A2.89,2.89,0,0,0,84.84,45.21Z"
      />
    </svg>
  );
}

Home.propTypes = {
  width: PropTypes.number,
  height: PropTypes.number,
  fill: PropTypes.string,
  className: PropTypes.string,
  classNamePath: PropTypes.string,
};

Home.defaultProps = {
  width: 40,
  height: 40,
  fill: 'currentColor',
  className: 'homeIcon',
  classNamePath: 'homeIconPath',
};

export default Home;
