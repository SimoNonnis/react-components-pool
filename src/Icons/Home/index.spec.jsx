/* eslint-env jest */
import React from 'react';

import Home from './';

describe('Test <Home /> icon', () => {
  it('should render Home icon correctly with default props', () => {
    const wrapper = shallow(<Home />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Home icon correctly with props', () => {
    const wrapper = shallow(<Home fill="#8A2BE2" height={30} width={30} />);
    expect(wrapper).toMatchSnapshot();
  });
});
