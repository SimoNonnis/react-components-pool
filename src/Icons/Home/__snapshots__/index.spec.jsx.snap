// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Home /> icon should render Home icon correctly with default props 1`] = `
<svg
  className="homeIcon"
  height={40}
  viewBox="0 0 100 100"
  width={40}
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    Home Icon
  </title>
  <path
    className="homeIconPath"
    d="M98.26,42.6a2.87,2.87,0,0,0-1.15-1.89L51.71,7.46a2.88,2.88,0,0,0-3.41,0L2.89,40.71A2.89,2.89,0,1,0,6.3,45.37l43.7-32,43.7,32a2.87,2.87,0,0,0,1.7.56,2.9,2.9,0,0,0,2.33-1.18A2.87,2.87,0,0,0,98.26,42.6Z"
    fill="currentColor"
  />
  <path
    className="homeIconPath"
    d="M84.84,45.21A2.89,2.89,0,0,0,82,48.1V87.32H62.84V66.09a12.85,12.85,0,0,0-25.69,0V87.32H18V48.1a2.89,2.89,0,0,0-5.78,0V90.21a2.89,2.89,0,0,0,2.89,2.89H40a2.9,2.9,0,0,0,2.88-2.65,2.26,2.26,0,0,0,0-.24V66.09a7.07,7.07,0,1,1,14.14,0V90.21a2.1,2.1,0,0,0,0,.22A2.9,2.9,0,0,0,60,93.1H84.84a2.89,2.89,0,0,0,2.89-2.89V48.1A2.89,2.89,0,0,0,84.84,45.21Z"
    fill="currentColor"
  />
</svg>
`;

exports[`Test <Home /> icon should render Home icon correctly with props 1`] = `
<svg
  className="homeIcon"
  height={30}
  viewBox="0 0 100 100"
  width={30}
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    Home Icon
  </title>
  <path
    className="homeIconPath"
    d="M98.26,42.6a2.87,2.87,0,0,0-1.15-1.89L51.71,7.46a2.88,2.88,0,0,0-3.41,0L2.89,40.71A2.89,2.89,0,1,0,6.3,45.37l43.7-32,43.7,32a2.87,2.87,0,0,0,1.7.56,2.9,2.9,0,0,0,2.33-1.18A2.87,2.87,0,0,0,98.26,42.6Z"
    fill="#8A2BE2"
  />
  <path
    className="homeIconPath"
    d="M84.84,45.21A2.89,2.89,0,0,0,82,48.1V87.32H62.84V66.09a12.85,12.85,0,0,0-25.69,0V87.32H18V48.1a2.89,2.89,0,0,0-5.78,0V90.21a2.89,2.89,0,0,0,2.89,2.89H40a2.9,2.9,0,0,0,2.88-2.65,2.26,2.26,0,0,0,0-.24V66.09a7.07,7.07,0,1,1,14.14,0V90.21a2.1,2.1,0,0,0,0,.22A2.9,2.9,0,0,0,60,93.1H84.84a2.89,2.89,0,0,0,2.89-2.89V48.1A2.89,2.89,0,0,0,84.84,45.21Z"
    fill="#8A2BE2"
  />
</svg>
`;
